modules/@apostrophecms/piece-type/index.js

```diff
@@ -1162,6 +1168,21 @@ module.exports = {
           };
           await self.insert(req, _new);
         }
+      },
+      disableBatchOperation(name, properties) {
+        const shouldDisablePublish = name === 'publish' && self.options.autopublish;
+        if (shouldDisablePublish) {
+          return true;
+        }
+
+        // Check if there is a required schema field for this batch operation.
+        const requiredFieldNotFound = properties.requiredField &&
+                !self.schema.some((field) => field.name === properties.requiredField);
+        if (requiredFieldNotFound) {
+          return true;
+        }
+
+        return false;
```

Comment:
This code is just about have a method that centralize disabling of some batch operations. And I removed `publish` batch operation in `autopublished` modules.

Reply:
Nice